fabric-common/lib/BlockDecoder.js

```diff
@@ -995,141 +1105,133 @@ function decodeConfigPolicy(proto_config_policy) {
+	implicit_meta_policy.sub_policy = implicitMetaPolicyProto.sub_policy;
+	implicit_meta_policy.rule = implicitMetaPolicyProto.rule;
+	implicit_meta_policy.ruleString = fabproto6.common.ImplicitMetaPolicy.Rule[implicitMetaPolicyProto.rule];
+
 	return implicit_meta_policy;
 }
 
-function decodeSignaturePolicyEnvelope(signature_policy_envelope_bytes) {
+function decodeSignaturePolicyEnvelope(signaturePolicyEnvelopeBuf) {
 	const signature_policy_envelope = {};
-	const proto_signature_policy_envelope = fabprotos.common.SignaturePolicyEnvelope.decode(signature_policy_envelope_bytes);
-	signature_policy_envelope.version = decodeVersion(proto_signature_policy_envelope.getVersion());
-	signature_policy_envelope.rule = decodeSignaturePolicy(proto_signature_policy_envelope.getRule());
-	const identities = [];
-	const proto_identities = proto_signature_policy_envelope.getIdentities();
-	if (proto_identities) {
-		for (const i in proto_identities) {
-			const msp_principal = decodeMSPPrincipal(proto_identities[i]);
-			identities.push(msp_principal); // string
+	const signaturePolicyEnvelopeProto = fabproto6.common.SignaturePolicyEnvelope.decode(signaturePolicyEnvelopeBuf);
+	signature_policy_envelope.version = convertVersion(signaturePolicyEnvelopeProto.version);
+	signature_policy_envelope.rule = decodeSignaturePolicy(signaturePolicyEnvelopeProto.rule);
+	signature_policy_envelope.identities = [];
+	const identitiesProto = signaturePolicyEnvelopeProto.identities;
+	if (identitiesProto) {
+		for (const mSPPrincipalProto of identitiesProto) {
+			const msp_principal = decodeMSPPrincipal(mSPPrincipalProto);
+			signature_policy_envelope.identities.push(msp_principal);
 		}
 	}
-	signature_policy_envelope.identities = identities;
 
 	return signature_policy_envelope;
 }
 
-function decodeSignaturePolicy(proto_signature_policy) {
+function decodeSignaturePolicy(signaturePolicyProto) {
 	const signature_policy = {};
-	signature_policy.Type = proto_signature_policy.Type;
-	if (signature_policy.Type === 'n_out_of') {
+	if (signaturePolicyProto.n_out_of) {
 		signature_policy.n_out_of = {};
-		signature_policy.n_out_of.N = proto_signature_policy.n_out_of.getN();
+		signature_policy.n_out_of.n = signaturePolicyProto.n_out_of.n;
 		signature_policy.n_out_of.rules = [];
-		for (const i in proto_signature_policy.n_out_of.rules) {
-			const proto_policy = proto_signature_policy.n_out_of.rules[i];
-			const policy = decodeSignaturePolicy(proto_policy);
-			signature_policy.n_out_of.rules.push(policy);
+		for (const childSignaturePolicyProto of signaturePolicyProto.n_out_of.rules) {
+			const child_signature_policy = decodeSignaturePolicy(childSignaturePolicyProto);
+			signature_policy.n_out_of.rules.push(child_signature_policy);
 		}
-	} else if (signature_policy.Type === 'signed_by') {
-		signature_policy.signed_by = proto_signature_policy.getSignedBy();
 	} else {
-		throw new Error('unknown signature policy type');
```

Comment:
This deletion inspire me that, we suffer a lot like this. 
There are some error handler which actually should be assertion, with cases rarely happen. 
There are many error catcher simply return a message wrapping original error message, without taking original error stack into concern

For both case it is better to expose those fatal error to developer. instead of handle them as take it for granted